### ISSUE TITLE
Have the pilot update the status of its PilotResource

### ISSUE THREAD

**Issue by wallrj** [jetstack/navigator#169]
Part of #140 

/kind bug

**Comment by wallrj** [comment]
Trying the Jolokia javaagent in order to access database status information via a REST API rather than having to use nodetool.

```
wget --quiet -O - http://localhost:8778/jolokia/list?maxDepth=1 | jq
wget https://github.com/rhuss/jolokia/releases/download/v1.3.7/jolokia-1.3.7-bin.tar.gz
tar xf jolokia-1.3.7-bin.tar.gz
-javaagent:/jolokia-1.3.7/agents/jolokia-jvm.jar
wget --quiet -O - http://localhost:8778/jolokia/read/org.apache.cassandra.db:type=StorageService/LiveNodes | jq .
```

See:
 * https://engineeringblog.yelp.com/2016/06/monitoring-cassandra-at-scale.html
 * https://github.com/apache/cassandra/blob/cassandra-3.11/src/java/org/apache/cassandra/service/StorageServiceMBean.java
 * https://jolokia.org/reference/html/protocol.html
 * http://blog.tomas.cat/en/2013/04/12/monitoring-cassandra-relevant-data-should-be-watched-and-how-send-it-graphite/
 * https://github.com/rhuss/jolokia/issues/256
 * https://medium.com/@mlowicki/cassandra-metrics-and-their-use-in-grafana-1f0dc33f9cca
 * https://medium.com/@mlowicki/monitoring-cassandra-a7fde5b2de9c

**Comment by munnerz** [comment]
I've taken a quick look at Jolokia & nodetool, although as you can imagine it's not something I've read much about before.

From what I can see, nodetool doesn't support output formats when you run a CLI command, so it'll take some nasty parsing to use it instead of some programmatic API that Jolokia provides.

I'm kind of hesitant however to set up/run Jolokia within each pod, simply because it's not something that we're familiar with yet, and given it's quite specific to Cassandra I think we could do with chatting to some experts to make sure that this *is* in fact the defacto for it. Is there any way we can introduce the Pilot resource type without this for now?

If this is something we can't do without, I think we should sync up next week and get on a call with someone that knows Cassandra well to verify our decisions 😄 wdyt?

**Comment by wallrj** [comment]
Fixed in #263
